### ISSUE TITLE
Inconsistent signatures in ITraceWriter between netframework and some netstandard builds

### ISSUE THREAD

**Issue by MarkusGeigerITD** [JamesNK/Newtonsoft.Json#1616]
Hi!

The ITraceWriter interface is defined in an inconsistent manner across the different build targets.
The TraceLevel enumeration is 
**System.Diagnostics.**TraceLevel for netframework and netstandard2.0 but  
**NewtonSoft.Json.**TraceLevel for <= netstandard1.6

That leads to an exception when a net461 project depends on a netstandard1.6 library that in turn depends on NewtonSoft.Json and implements a custom trace writer (--> TypeLoadException when loading the custom trace writer)

**Comment by JamesNK** [comment]
Yeah I know. There use to be a lot more of this problem, and ITraceWriter is the last place where it happens. If you're writing a library and you have an ITraceWriter then you'll need to multi target netstandard and .NET Framework

**Comment by jogibear9988** [comment]
Exists there a Solution for this?

**Comment by lgolding** [comment]
@jogibear9988 Here's what I did:

In my "build.props" file (which I import into every .csproj file):
```
  <PropertyGroup Label="ConditionalCompilation">
    <DefineConstants Condition=" '$(TargetFramework)' == 'net461' ">$(DefineConstants);NET461</DefineConstants>
  </PropertyGroup>
```
In every source file that uses `TraceLevel`:
```
// Newtonsoft defines TraceLevel inconsistently between .NET Framework and .NET Standard.
// This fixes the problem. See https://github.com/JamesNK/Newtonsoft.Json/issues/1616.
#if NET461
using TraceLevel = System.Diagnostics.TraceLevel;
#else
using TraceLevel = Newtonsoft.Json.TraceLevel;
#endif

```
